ui/src/app/utils/index.ts

```diff
@@ -152,39 +151,40 @@ export function cookiesEnabled(): boolean {
     }
 }
 
-export const ReactComponent = ({ propNames = [], ...options }) => WrappedComponent => {
-  class WrapperComponent implements DoCheck, OnInit, OnDestroy {
-    @ViewChild('root')
-    rootElement: ElementRef;
+/**
+ * Helper base class for defining an Angular-wrapped React component. This is a
+ * stop-gap for React migration.
+ *
+ * Requirements:
+ *  - Component template must contain a div labeled "#root".
+ *  - React propNames must exactly match instance property names on the subclass
+ *    (usually these are also annotated as Angular @Inputs)
+ */
+export class ReactWrapperBase implements DoCheck, OnInit, OnDestroy {
+  @ViewChild('root') rootElement: ElementRef;
 
-    ngOnInit(): void {
-      this.renderComponent();
-    }
+  constructor(private wrapped: new (...args: any[]) => React.Component,
```

Comment:
Is this the right type? This need to work with both 'class' and 'function' components.

Reply:
Thanks, wasn't thinking about that - according to my local testing this should do the trick.